src/odb/src/lefin/lefTechLayerAreaRuleParser.cpp

```diff
@@ -0,0 +1,203 @@
+    odb::dbTechLayerAreaRule* rule)
+{
+  std::pair<int, int> size(lefin_->dbdist(at_c<0>(params)),
+                           lefin_->dbdist(at_c<1>(params)));
+  rule->setExceptEdgeLengths(size);
+}
+
+void lefTechLayerAreaRuleParser::setExceptMinSize(
+    const boost::fusion::vector<double, double>& params,
+    odb::dbTechLayerAreaRule* rule)
+{
+  std::pair<int, int> size(lefin_->dbdist(at_c<0>(params)),
+                           lefin_->dbdist(at_c<1>(params)));
+  rule->setExceptMinSize(size);
+}
+
+void lefTechLayerAreaRuleParser::setExceptStep(
+    const boost::fusion::vector<double, double>& params,
+    odb::dbTechLayerAreaRule* rule)
+{
+  std::pair<int, int> size(lefin_->dbdist(at_c<0>(params)),
+                           lefin_->dbdist(at_c<1>(params)));
+  rule->setExceptStep(size);
+}
+
+void lefTechLayerAreaRuleParser::setTrimLayer(
+    std::string val,
+    odb::dbTechLayerAreaRule* rule,
+    odb::dbTechLayer* layer,
+    std::vector<std::pair<odb::dbObject*, std::string>>& incomplete_props)
+{
+  auto trim_layer = layer->getTech()->findLayer(val.c_str());
+  if (trim_layer != nullptr) {
+    rule->setTrimLayer(trim_layer);
+  } else {
+    incomplete_props.push_back({rule, val});
+  }
+}
+
+bool lefTechLayerAreaRuleParser::parseSubRule(
+    std::string s,
+    odb::dbTechLayer* layer,
+    std::vector<std::pair<odb::dbObject*, std::string>>& incomplete_props)
+{
+  qi::rule<std::string::iterator, std::string(), ascii::space_type> _string;
+  _string %= lexeme[+(char_ - ' ')];
+  odb::dbTechLayerAreaRule* rule = odb::dbTechLayerAreaRule::create(layer);
+
+  qi::rule<std::string::iterator, space_type> EXCEPTEDGELENGTH
+      = ((lit("EXCEPTEDGELENGTH") >> double_ >> double_)[boost::bind(
+             &lefTechLayerAreaRuleParser::setExceptEdgeLengths, this, _1, rule)]
+         | lit("EXCEPTEDGELENGTH") >> double_[boost::bind(
+               &lefTechLayerAreaRuleParser::setInt,
+               this,
+               _1,
+               rule,
+               &dbTechLayerAreaRule::setExceptEdgeLength)]);
+
+  qi::rule<std::string::iterator, space_type> EXCEPTMINSIZE
+      = (lit("EXCEPTMINSIZE") >> double_ >> double_)[boost::bind(
+          &lefTechLayerAreaRuleParser::setExceptMinSize, this, _1, rule)];
+
+  qi::rule<std::string::iterator, space_type> EXCEPTSTEP
+      = (lit("EXCEPTSTEP") >> double_ >> double_)[boost::bind(
+          &lefTechLayerAreaRuleParser::setExceptStep, this, _1, rule)];
+
+  qi::rule<std::string::iterator, space_type> LAYER
+      = ((lit("LAYER")
+          >> _string[boost::bind(&lefTechLayerAreaRuleParser::setTrimLayer,
+                                 this,
+                                 _1,
+                                 rule,
+                                 layer,
+                                 boost::ref(incomplete_props))])
+         >> lit("OVERLAP")
+         >> double_[boost::bind(&lefTechLayerAreaRuleParser::setNumber,
```

Comment:
Why double_? I think this should be int_
Also I don't think there is a need for setNumber as it's just passing  the number to the function without any processing. you could use setOverlap directly.